CMakeLists.txt

```diff
@@ -235,6 +235,11 @@ if(MSVC)
 
     # Prevent warnings abount non-existing .pdb files
     add_link_options($<$<CONFIG:Debug>:/ignore:4099>)
+
+    if (CMAKE_SIZEOF_VOID_P EQUAL 4)
+        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /LARGEADDRESSAWARE")
+        set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /LARGEADDRESSAWARE")
+    endif()
```

Comment:
What's the context behind these changes? Will SDKs need to adopt the same settings?

Reply:
no, not really. From what i understand, this essentially raises virtual memory limit from 2gb to full 4 for 32bit app, and this old behavior is the default on window for compatibility reasons, but is not in any way relevant for modern apps. In practise, i was checking if under some stress our tests pass with UNITTEST_THREADS=128 on 8 core system with enough memory. Virtual memory peaks in realm-tests and realm-sync-tests at about ~3.5gb, so this just makes it possible to run with these conditions. Anyway, it shouldn't have any effect on anything else other than our test binaries.